### PR TITLE
[Reshard] Support partial to shard

### PR DESCRIPTION
**Pull request by LiYuRio**
### PR types

New features

### PR changes

Others

### Description

支持partial状态到shard状态的转换
Pcard-73145

### CODE REVIEW

File: paddle/phi/core/distributed/auto_parallel/reshard/p_to_s_reshard_function.cc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                            dtype,
+                            in_process_ids,
+                            in_reduce_scatter,
+                            static_cast<int64_t>(in_process_ids.size()),
+                            &out_reduce_scatter);
+
+  if (out_split_axis != 0) {
+    std::vector<int> axis;
+    for (size_t i = 1; i < vectorize(logical_ddim).size(); ++i) {
+      axis.emplace_back(i);
```
Comment (ForFishes) [inline paddle/phi/core/distributed/auto_parallel/reshard/p_to_s_reshard_function.cc]: Whether the axis here is the same as the previous one, no additional calculation is required.

Comment (LiYuRio) [reply paddle/phi/core/distributed/auto_parallel/reshard/p_to_s_reshard_function.cc]: Excellent! You are totally right!

**Review by ForFishes — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for ` set_tests_properties(test_reshard_p_to_s
                       PROPERTIES LABELS "RUN_TYPE=EXCLUSIVE" TIMEOUT 100)`

**Pull request merged**